### PR TITLE
docs(examples): added file upload example

### PR DESCRIPTION
**Pull request by marvinwu**
# upload file and images

This is a simple example of using the remix buildin [uploadHandler](https://remix.run/docs/en/v1/api/remix#uploadhandler) and Form with multipart data to upload an image file and display it,
it also show a simple(though not efficient way) of integrate with cloudinary without writing custom handler.

the relevent files are:

```
├── app
│   ├── routes
│   │   ├── cloudinary-upload.tsx // upload to cloudinary
│   │   └── local-upload.tsx // local upload using build in [createfileuploadhandler](https://remix.run/docs/en/v1/api/remix#unstable_createfileuploadhandler)
│   └── utils
│       └── utils.server.ts  // init cloudinary nodejs client on server side
|── .env // holds cloudinary credentails
```

## steps to set up cloudinary

- sign up a free [cloudinary account](https://cloudinary.com/)
- get the cloudname, api key and api secret from dashboard
- copy the .env.sample to .env and populate the credentials

### CODE REVIEW

**Comment by ryanflorence** [comment]
@kentcdodds can you review this example and merge when you're ready?

File: examples/file-and-cloudinary-upload/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- copy the .env.sample to .env and populate the credentials
+
+## caveat
+
+- upload large file is not seems stable now, try upload image with size < 30kb
+
+Open this example on [CodeSandbox](https://codesandbox.com):
+
+<!-- TODO: update this link to the path for your example: -->
+
```
Comment (kentcdodds) [inline examples/file-and-cloudinary-upload/README.md:31]:
```suggestion
```

File: examples/file-and-cloudinary-upload/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+## steps to set up cloudinary
+
+- sign up a free [cloudinary account](https://cloudinary.com/)
+- get the cloudname, api key and api secret from dashboard
+- copy the .env.sample to .env and populate the credentials
+
+## caveat
+
+- upload large file is not seems stable now, try upload image with size < 30kb
+
```
Comment (kentcdodds) [inline examples/file-and-cloudinary-upload/README.md:27]: @jacob-ebey, is this still the case or did you fix this recently?

Can we remove it?

```suggestion
```

File: examples/file-and-cloudinary-upload/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+<!-- TODO: update this link to the path for your example: -->
+
+[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/remix-run/remix/tree/main/examples/file-and-cloudinary-upload)
+
+## Example
+
+## Related Links
+
+Link to documentation or other related examples.
```
Comment (kentcdodds) [inline examples/file-and-cloudinary-upload/README.md:38]: @marvinwu could you add links to documentation here please?

File: examples/file-and-cloudinary-upload/app/routes/cloudinary-upload.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+type ActionData = {
+  errorMsg?: string;
+  imgSrc?: string;
+};
+
+export const action: ActionFunction = async ({ request }) => {
+  const uploadHandler = unstable_createFileUploadHandler({
+    maxFileSize: 30000
+  });
```
Comment (kentcdodds) [inline examples/file-and-cloudinary-upload/app/routes/cloudinary-upload.tsx:19]: Rather that saving the image to disk first and then uploading it like we're doing here, we should try to get streaming working (as demonstrated here: https://remix.run/docs/en/v1/api/remix#custom-uploadhandler).

Could you try to get that working so we can demonstrate that instead? It's more optimized and realistic.

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Looking good! I've got a few bits of feedback for this one.

File: examples/file-and-cloudinary-upload/app/routes/cloudinary-upload.tsx (hunk 4 above)
Comment (marvinwu) [reply examples/file-and-cloudinary-upload/app/routes/cloudinary-upload.tsx:19]: sure will give a try.

File: examples/file-and-cloudinary-upload/README.md (hunk 3 above)
Comment (marvinwu) [reply examples/file-and-cloudinary-upload/README.md:36]: @kentcdodds thx much for reviewing! sure will update it with doc links

Comment (marvinwu) [reply examples/file-and-cloudinary-upload/README.md]: @kentcdodds fixed, kindly take a look 

[updated readme](https://github.com/remix-run/remix/commit/072560795c54b229f2cad52776e16b09af10283e#diff-38bd30ea75140e0835f0538470e537a5be357755a5719294833d20bc10a748f0)

File: examples/file-and-cloudinary-upload/README.md (hunk 2 above)
Comment (marvinwu) [reply examples/file-and-cloudinary-upload/README.md]: then i will removed this caveat from the example readme

File: examples/file-and-cloudinary-upload/app/root.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width,initial-scale=1" />
+        <Meta />
+        <Links />
+      </head>
+      <body>
+        <Outlet />
+        <ScrollRestoration />
+        <Scripts />
+        {process.env.NODE_ENV === "development" && <LiveReload />}
```
Comment (MichaelDeBoey) [inline examples/file-and-cloudinary-upload/app/root.tsx:23]:
```suggestion
        <LiveReload />
```
See #1352

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: examples/file-and-cloudinary-upload/app/routes/local-upload.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    uploadHandler
+  );
+  const image = formData.get("img");
+  if (!image) {
+    return json({
+      error: "something wrong"
+    });
+  }
+  return json({
+    imgSrc: image.name
```
Comment (josuevalrob) [inline examples/file-and-cloudinary-upload/app/routes/local-upload.tsx:31]: Property 'name' does not exist on type 'FormDataEntryValue'.
https://stackoverflow.com/questions/71090990/typescript-property-name-does-not-exist-on-type-formdataentryvalue

File: contributors.yml [hunk 7]
```diff
@@ -225,4 +226,3 @@
 - yesmeck
 - yomeshgupta
 - zachdtaylor
-- zainfathoni
```
Comment (kentcdodds) [inline contributors.yml:228]: I'm guessing this was in error

File: contributors.yml [hunk 8]
```diff
@@ -225,4 +226,3 @@
 - yesmeck
 - yomeshgupta
 - zachdtaylor
```
Comment (kentcdodds) [inline contributors.yml:228]:
```suggestion
- zachdtaylor
- zainfathoni
```

File: examples/file-and-cloudinary-upload/README.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+[![Open in CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/github/remix-run/remix/tree/main/examples/file-and-cloudinary-upload)
+
+## Example
+
+## Related Links
+
+### Remix Documentation
+
+- [Handle Multiple Part Forms(File Uplodas)](https://remix.run/docs/en/v1/api/remix#unstable_parsemultipartformdata-node)
```
Comment (kentcdodds) [inline examples/file-and-cloudinary-upload/README.md:35]:
```suggestion
- [Handle Multiple Part Forms (File Uploads)](https://remix.run/docs/en/v1/api/remix#unstable_parsemultipartformdata-node)
```

**Review by kentcdodds — APPROVED** [review]
Sweet! This is very cool 👍

**Comment by kentcdodds** [comment]
Thanks everyone 👏👏

File: examples/file-and-cloudinary-upload/app/utils/utils.server.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import dotenv from "dotenv";
+import cloudinary from "cloudinary";
+import type { Stream } from "stream";
+
+dotenv.config();
```
Comment (silvenon) [inline examples/file-and-cloudinary-upload/app/utils/utils.server.ts:5]: Love this example! Just a question about environment variables: according to the docs `dotenv` should only be used when starting the development script via `node -r` (see [server environment variables](https://remix.run/docs/en/v1/guides/envvars#server-environment-variables)). However, I prefer your way much better, is there something wrong with it?

Comment (jacob-ebey) [reply examples/file-and-cloudinary-upload/app/utils/utils.server.ts:5]: I always avoid calling it via code. It's a potential source of bugs if you ship a ".env" to production by accident.

Comment (kentcdodds) [reply examples/file-and-cloudinary-upload/app/utils/utils.server.ts:5]: Agreed! Fixed 👍

**Pull request merged**